src/platform/tests/MpiServerTests.cpp

```diff
@@ -0,0 +1,243 @@
+    {
+        char* response = nullptr;
+        int responseSize = 0;
+
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_CLOSE_URI, "{\"ClientSession\": 123}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_CLOSE_URI, "{}", &response, &responseSize, g_mpiCalls));
+
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": 123, \"ComponentName\": \"\", \"ObjectName\": \"\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ComponentName\": \"\", \"ObjectName\": \"\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"\", \"ComponentName\": 123, \"ObjectName\": \"\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"\", \"ObjectName\": \"\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"\", \"ComponentName\": \"\", \"ObjectName\": 123, \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"\", \"ComponentName\": \"\", \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_URI, "{\"ClientSession\": \"\", \"ComponentName\": \"\", \"ObjectName\": \"\"}", &response, &responseSize, g_mpiCalls));
+
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_URI, "{\"ClientSession\": 123, \"ComponentName\": \"\", \"ObjectName\": \"\"}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_URI, "{\"ComponentName\": \"\", \"ObjectName\": \"\"}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_URI, "{\"ClientSession\": \"\", \"ComponentName\": 123, \"ObjectName\": \"\"}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_URI, "{\"ClientSession\": \"\", \"ObjectName\": \"\"}", &response, &responseSize, g_mpiCalls));
+
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"ClientSession\": 123, \"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"Payload\": {}}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_SET_DESIRED_URI, "{\"ClientSession\": \"\"}", &response, &responseSize, g_mpiCalls));
+
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_REPORTED_URI, "{\"ClientSession\": 123}", &response, &responseSize, g_mpiCalls));
+        EXPECT_EQ(HTTP_BAD_REQUEST, HandleMpiCall(MPI_GET_REPORTED_URI, "{}", &response, &responseSize, g_mpiCalls));
```

Comment:
>EXPECT_EQ(H [](http://example.com/codeflow?start=8&length=11)

Add  FREE_MEMORY(response); to this and all else as well in case there is allocated

Reply:
Or add validation that returned responses are NULL and sizes 0